interfaces/builtin/system_packages_doc.go

```diff
@@ -41,6 +41,7 @@ const systemPackagesDocConnectedPlugAppArmor = `
 
 /usr/share/doc/{,**} r,
 /usr/share/libreoffice/help/{,**} r,
+/usr/share/xubuntu-docs/{,**} r,
```

Comment:
Bit unhappy that we have another special snowflake, as if the original packagers/upstream devs think /usr/share/doc is just for fun.

Reply:
That's actually per [Debian packaging policy](https://www.debian.org/doc/debian-policy/ch-docs.html#additional-documentation):

> Packages must not require the existence of any files in `/usr/share/doc/` in order to function. Any files that are used or read by programs but are also useful as stand alone documentation should be installed elsewhere, such as under `/usr/share/package/`, and then included via symbolic links in `/usr/share/doc/package`.

Reply:
In jammy I can see 289 packages which seem to ship docs under `/usr/share/xxxx-docs?/` - should we try and be more inclusive here and not have a heap of special snowflakes - https://pastebin.ubuntu.com/p/JfYXBXW7Nk/

e.g. I think the following may capture all of these:

```
/usr/share/*-doc{,s}/{,**} r,
```

Reply:
By the way, should we remove the comma from the patterns `{,**}` in both the existing rule `/usr/share/doc/{,**} r,` and the one you proposed just now? Otherwise we are also giving out the information of which packages are installed on the system, which might be an unnecessary privacy leak.

Reply:
@alexmurray very good suggestion, I'm mildly ashamed I didn't think of it myself. So you figured out the apparmor rule, how do we go about the mount specs? Do we need to iterate over all the relevant docs directories under `/usr/share/` on the host and emit the corresponding mount instructions and entries? I'm not familiar with the internals of interfaces, having only submitted trivial patches until now, so I could use some guidance here.

Reply:
@mardy I think the comma in the patterns is intended, to allow a user to conveniently browse all docs by navigating to `file:///usr/share/doc`. Otherwise they would have to remember the package name to access its doc from within the browser.

Reply:
> That's actually per [Debian packaging policy](https://www.debian.org/doc/debian-policy/ch-docs.html#additional-documentation):
> 
> > Packages must not require the existence of any files in `/usr/share/doc/` in order to function. Any files that are used or read by programs but are also useful as stand alone documentation should be installed elsewhere, such as under `/usr/share/package/`, and then included via symbolic links in `/usr/share/doc/package`.

My opinion is that that interpretation of Debian Policy shouldn't apply for any package name that ends in `doc` or `docs`. There are a lot of `-doc` packages that wouldn't "function" if the system was set to path-exclude /usr/share/doc/ but that is expected.

LibreOffice is a special case because it's a non-doc package where the Help button won't work if the docs aren't there.

Reply:
Hi, Xubuntu developer here -- sorry for the noise.

In the interest of having our docs working on the 22.04 release in less than two weeks, what are the next steps? Will snap be updated to support the `/usr/share/xubuntu-docs` path, or does the `xubuntu-docs` packaging need to be updated to install to a different directory? Thanks!

Reply:
Proposed the boarded AA rule in #11699